TensorLib/Tensor.lean

```diff
@@ -0,0 +1,135 @@
+import Mathlib.Tactic
+namespace TensorLib
+section TensorData
+
+variable (a: Type)
+
+-- I tried `td_node (x: TensorData) (xs: List TensorData)` to remove the redundant empty node
+-- from the type. (We can already represent the 0 shape with `td_root []`.)
+-- This design works, but this way the examples are more uniform; it looked weird to
+-- syntactically preference the first of every node list. The code has some annoying special
+-- cases for the empty node now, though, so it may be worth revisiting this.
+inductive TensorData where
```

Comment:
Should we call this `Tensor`? That's how we refer to this datatype, and that's the name of the file.

Reply:
Was thinking that the tensor would have some metadata in it, e.g. the shape.